pennylane/ops/functions/equal.py

```diff
@@ -168,6 +169,38 @@ def _equal(
     raise NotImplementedError(f"Comparison of {type(op1)} and {type(op2)} not implemented")
 
 
+@_equal.register
+def _equal_circuit(
+    op1: qml.tape.QuantumScript,
+    op2: qml.tape.QuantumScript,
+    check_interface=True,
+    check_trainability=True,
+    rtol=1e-5,
+    atol=1e-9,
+):
+    # operations
+    if len(op1.operations) != len(op2.operations):
+        return False
+    for comparands in itertools.zip_longest(op1.operations, op2.operations):
+        if not qml.equal(
+            comparands[0],
+            comparands[1],
+            check_interface=check_interface,
+            check_trainability=check_trainability,
+            rtol=rtol,
+            atol=atol,
+        ):
+            return False
+    # measurements
+    if op1.measurements != op2.measurements:
```

Comment:
I would suggest using a loop and also adding checks for interfaces and trainability for the measurements. Observables such as `Hamiltonian`, `Hermitian`, `Sum`, `Prod` can have numerical parameters.

Reply:
feel free to resolve this @mudit2812 :)